packages/l10n/README.md

```diff
@@ -97,3 +97,265 @@ First, download the data (`core.zip`) for a specific version from the [downloads
+
+```js
+const product = {
+  name: {
+    en: 'Milk',
+    de: 'Milch',
+  },
+};
+```
+
+When rendering a Product in a view, we would like to render the value of a localized string field, such as `name`, given the selected **data locale** of the UI.
+The **data locale** is a value controlled by the Merchant Center User (MC User) by changing the **data locale switcher** in the top bar of the Merchant Center. The list of available options is derived by the list of languages specified in the project. The selected value can be read from the **application context**.
+
+However, there might be a case where the selected **data locale** does not match any of the localized string values. In this case, it would be helpful to display a "fallback" value using the `formatLocalizedString` function.
+
+The `formatLocalizedString` util is authored with a couple of things in mind:
+
+1. Help us Custom Application developers remain resilient as we attempt to derive a value of `LocalizedString` given a specified locale.
+2. Help us Custom Application developers remain consistent when rendering a value derived from `LocalizedString`.
+
+Let's take a look at the examples below to put this to action.
+
+#### Example usage
+
+All examples below will use the following `Product`:
+
+```
+const product = {
+  name: {
+    en: 'Milk',
+    de: 'Milch,
+  }
+}
+```
+
+##### Scenario 1
+
+Given that selected **data locale** of `de`
+
+```js
+const translatedName = formatLocalizedString(product, {
+  key: 'name',
+  locale: 'de',
+});
+
+console.log(translatedName);
+// 'Milch'
+```
+
+##### Scenario 2
+
+Given that selected **data locale** of `sv`
+
+```js
+const translatedName = formatLocalizedString(product, {
+  key: 'name',
+  locale: 'sv',
+});
+
+console.log(translatedName);
+// 'Milk (EN)'
+```
+
+**What happened?**
+
+Our product has no value for the selected **data locale** `sv`. The `formatLocalizedString` function selects the "next available value" from `product.name`, in this case `en`, and returns it with a hint `(EN)` that this value refers to another locale.
+
+##### Scenario 3
+
+Given that selected **data locale** of `de-AT`
+
+```js
+const translatedName = formatLocalizedString(product, {
+  key: 'name',
+  locale: 'de-AT',
+});
+
+console.log(translatedName);
+// 'Milch'
+```
+
+**What happened?**
+
+Our product has no matching value for the selected **data locale** `de-AT` but we still got `"Milch"`.
+This is because `formatLocalizedString` attempts to determine a **primary language tag** on the option `locale` and match that to the available values. Given that `de` is a [primary](https://en.wikipedia.org/wiki/IETF_language_tag) of `de-AT`, it determines that this is the _closest available value_ from `product.name`
+
+#### Fallback
+
+Given that `formatLocalizedString` can not handle all scenario exemplified above, we can specify a `fallback` as a last resort:
```

Comment:
Got confused by "not all cases above". I felt it handles all cases we talk about above. So for me this here is another case we tackle. Hence a suggestion if I got that flow right. 

```suggestion
To provide even more freedom beyond cases mentioned above the `formatLocalizedString` allows specifying a `fallback` as a last resort:
```